bitcoin/src/address/mod.rs

```diff
@@ -631,10 +631,12 @@ impl Address {
 /// Methods that can be called only on `Address<NetworkUnchecked>`.
```

Comment:
I also noticed the doc on top of `Address` mentions `AddressInner` which is a private type so it's not suitable for public doc. Should just say `#[repr(transparent)]` is a private detail, no need to explain why publicly. We can have a non-doc comment that it's to allow these casts.